src/sentry/static/sentry/app/views/taskList/tasks.jsx

```diff
@@ -0,0 +1,123 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+import {t} from 'app/locale';
+import {tasksGet} from 'app/redux/actions/task';
+import {Panel, PanelBody} from 'app/components/panels';
+import ProcessListItem from 'app/components/task/processListItem';
+import LoadingError from 'app/components/loadingError';
+import LoadingIndicator from 'app/components/loadingIndicator';
+
+class Tasks extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    const {getTasks} = this.props;
+    getTasks();
+  }
+
+  renderBody() {
+    const {tasks, loading, errorMessage, getTasks} = this.props;
+
+    let body;
+    if (loading) {
+      body = this.renderLoading();
+    } else if (errorMessage) {
+      body = <LoadingError message={errorMessage} onRetry={getTasks} />;
+    } else if (tasks.length > 0) {
+      body = this.renderProcesses();
+    } else {
+      body = this.renderEmpty();
+    }
+    return body;
+  }
+
+  groupTasksByProcess() {
```

Comment:
I wonder if this could be achieved in shorter code with more functional style (reducer), similar to what this person is doing: https://medium.com/@edisondevadoss/javascript-group-an-array-of-objects-by-key-afc85c35d07e